pwnlib/util/fiddling.py

```diff
@@ -971,3 +973,116 @@ def bnot(value, width=None):
         width = context.bits
     mask = ((1<<width)-1)
     return mask ^ value
+
+@LocalNoarchContext
+def js_escape(data, padding=context.cyclic_alphabet[0:1], **kwargs):
+    r"""js_escape(data, padding=context.cyclic_alphabet[0:1], endian = None, **kwargs) -> str
+
+    Pack data as an escaped Unicode string for use in JavaScript's `unescape()` function
+
+    Arguments:
+        data (bytes): Bytes to pack
+        padding (bytes): A single byte to use as padding if data is of uneven length
+        endian (str): Endianness with which to pack the string ("little"/"big")
+
+    Returns:
+        A string representation of the packed data
+
+    >>> js_escape(b'\xde\xad\xbe\xef')
+    '%uadde%uefbe'
+
+    >>> js_escape(b'\xde\xad\xbe\xef', endian='big')
+    '%udead%ubeef'
+
+    >>> js_escape(b'\xde\xad\xbe')
+    '%uadde%u61be'
+
+    >>> js_escape(b'aaaa')
+    '%u6161%u6161'
+    """
+    data = packing._need_bytes(data)
+
+    if len(padding) != 1:
+        raise ValueError("Padding must be a single byte")
+    padding = packing._need_bytes(padding)
+
+    if len(data) % 2:
+        data += padding[0:1]
+
+    if isinstance(data, six.string_types):
+        # Give Python 2 an iterable of ints, similar to how a Python 3 bytes works
+        data = map(ord, data)
```

Comment:
There is a universal cross-version way to handle this:
```suggestion
    data = bytearray(data)
```
not sure, but it is possible that we do it already in _need_bytes.